### PR TITLE
Remove checking the resource key with the annotation because their generation logic are different

### PR DESCRIPTION
**Pull request by ffjlabo**
**What this PR does / why we need it**:

context: https://github.com/pipe-cd/pipecd/issues/4269#issuecomment-2031079364

Removed checking the resource key with the annotation because their generation logic differs.

Previously, I tried to keep using the annotation `pipecd.dev/resource-key`, but found that some k8s resources inherit the parent's annotation and can't be attached correct resource key.
ref: https://github.com/pipe-cd/pipecd/pull/4858#issuecomment-2048982687

So I decided to remove the checking.

**Which issue(s) this PR fixes**:

Part of https://github.com/pipe-cd/pipecd/issues/4269
Previous PR https://github.com/pipe-cd/pipecd/pull/4858

**Does this PR introduce a user-facing change?**:

- **How are users affected by this change**:
- **Is this breaking change**:
- **How to migrate (if breaking change)**:

### CODE REVIEW

**Comment by ffjlabo** [comment]
## cluster scoped resource

- ✅ namespace exists in app.pipecd.yaml　
- ✅ namespace doesn't exist in app.pipecd.yaml

## namespace scoped resource

- ✅ namespace exists in app.pipecd.yaml　
- ✅ namespace doesn't exist in app.pipecd.yaml

**Comment by ffjlabo** [comment]
There is a problem about the resource key of cluster scoped resources. I'll consider it later to modify the effect scope.

File: pkg/app/piped/platformprovider/kubernetes/applier.go [hunk 1]
```diff
@@ -147,21 +146,17 @@ func (a *applier) Delete(ctx context.Context, k ResourceKey) (err error) {
 		return a.initErr
 	}
 
-	m, err := a.kubectl.Get(
+	// Check the resource exists or not using the resource key.
```
Comment (t-kikuc) [inline pkg/app/piped/platformprovider/kubernetes/applier.go:149]:
```suggestion
	// Check whether the resource exists or not using the resource key.
```

**Comment by ffjlabo** [comment]
close this PR because I decided another solution for it. 
context: https://github.com/pipe-cd/pipecd/issues/4269#issuecomment-2104055928
follow the PR → https://github.com/pipe-cd/pipecd/pull/4916

**Pull request closed without merging**